### PR TITLE
Update change password

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

**Comment by itkg-rboessel** [comment]
https://github.com/open-orchestra/open-orchestra/pull/475
https://github.com/open-orchestra/open-orchestra-user-bundle/pull/51

File: UserAdminBundle/Controller/Admin/UserController.php [hunk 1]
```diff
@@ -71,4 +72,37 @@ public function formAction(Request $request, $userId)
...
+        /* @var User $user */
+        $user = $this->get('open_orchestra_user.repository.user')->find($userId);
+        $form = $this->createForm('user_change_password', $user, array(
+            'action' => $this->generateUrl('open_orchestra_user_admin_user_change_password', array('userId' => $userId))
+        ));
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $userManager = $this->get('fos_user.user_manager');
+            $userManager->updateUser($user, true);
```
Comment (nicolasThal) [inline UserAdminBundle/Controller/Admin/UserController.php]: you should not let the manager du the flush, it should be done once it for all at the end of the process

File: UserAdminBundle/Controller/Admin/UserController.php [hunk 2]
```diff
@@ -66,9 +67,41 @@ public function formAction(Request $request, $userId)
...
+    public function changePasswordAction(Request $request, $userId)
+    {
+        /* @var User $user */
+        $user = $this->get('open_orchestra_user.repository.user')->find($userId);
+        $form = $this->createForm('user_change_password', $user, array(
+            'action' => $this->generateUrl('open_orchestra_user_admin_user_change_password', array('userId' => $userId))
+        ));
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
```
Comment (nicolasThal) [inline UserAdminBundle/Controller/Admin/UserController.php]: You should also dispatch an event linked to the change password

**Comment by nicolasThal** [comment]
what happen when you submit a form?

**Comment by itkg-rboessel** [comment]
@nicolasThal @itkg-ngilain @itkg-nanne

File: BackofficeBundle/Resources/public/coffee/FullPagePanelView.coffee [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      self_version: element.get('links')._self_version
+      path: 'showEntityWithLanguageAndVersion'
+    ) if element.get('links')._self_version
+
+    @options = $.extend(@options, duplicate:
+      language: element.get('language')
+      self_duplicate: element.get('links')._self_duplicate
+      path: 'showEntityWithLanguage'
+    ) if element.get('links')._self_duplicate
+)
```
Comment (itkg-nanne) [inline BackofficeBundle/Resources/public/coffee/FullPagePanelView.coffee]: to much duplicate code from FullPageFormView. Maybe could you use $.extend?

File: BackofficeBundle/Resources/public/coffee/table/tableviewLoader.coffee [hunk 4]
```diff
@@ -46,8 +46,13 @@ tableViewLoad = (link, entityType, entityId, language, version, add, sourceLangu
               link = link + '&version=' + version if version != undefined
               founded = tableViewLoadSpecificElement(link, title, entityType)
```
Comment (itkg-nanne) [inline BackofficeBundle/Resources/public/coffee/table/tableviewLoader.coffee]: This method call fullPageFormView, your 'else' too.

File: UserAdminBundle/Controller/Admin/UserController.php [hunk 5]
```diff
@@ -49,7 +50,7 @@ public function newAction(Request $request)
 
     /**
      * @param Request $request
-     * @param string  $userId
+     * @param string $userId
```
Comment (itkg-nanne) [inline UserAdminBundle/Controller/Admin/UserController.php]: parameter no more aligned

File: UserAdminBundle/Form/Type/ChangePasswordUserType.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace OpenOrchestra\UserAdminBundle\Form\Type;
+
+use Symfony\Component\Form\FormBuilderInterface;
+use FOS\UserBundle\Form\Type\ChangePasswordFormType;
+use OpenOrchestra\BackofficeBundle\EventSubscriber\AddSubmitButtonSubscriber;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ChangePasswordUserType extends ChangePasswordFormType
```
Comment (nicolasThal) [inline UserAdminBundle/Form/Type/ChangePasswordUserType.php]: This form type should be in the UserBundle I think

**Comment by itkg-rboessel** [comment]
https://github.com/open-orchestra/open-orchestra-user-bundle/pull/53

File: BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee [hunk 7]
```diff
@@ -2,21 +2,21 @@ FullPageFormView = OrchestraView.extend(
   el: '#content'
 
   initialize: (options) ->
+    @initializer options
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee]: those modification will impact @juchi, is he ok with it?

File: BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee [hunk 8]
```diff
@@ -2,21 +2,21 @@ FullPageFormView = OrchestraView.extend(
...
+    ]
+    return
+
+  initializer: (options) ->
     @options = options
     @options.listUrl = appRouter.generateUrl('listEntities', entityType: options.entityType) if options.listUrl == undefined
     @completeOptions(@options.element) if @options.element != undefined
     @events = @events || {}
-    @loadTemplates [
-      'OpenOrchestraBackofficeBundle:BackOffice:Underscore/fullPageFormView',
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee]: I don't understand : you remove the loading of the fullPageFormView template from the fullPageForView backbone view ? Is it what you want or a mistake ?

File: BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee [hunk 9]
```diff
@@ -2,21 +2,21 @@ FullPageFormView = OrchestraView.extend(
...
-    ]
-    return
 
   render: ->
-    if @options.multiVersion
-      @options.title = @renderTemplate('OpenOrchestraBackofficeBundle:BackOffice:Underscore/elementTitle',
-        element: @options.element
-      )
+    @callMultiVersionOptions()
     $(@el).html(@renderTemplate('OpenOrchestraBackofficeBundle:BackOffice:Underscore/fullPageFormView', @options))
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee]: On my mind, this won't work as the template is no more loaded on the initialization

File: BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee (hunk 7 above)
Comment (juchi) [inline BackofficeBundle/Resources/public/coffee/FullPageFormView.coffee]: I'm actually not impacted (I'm not overloading the initilize method yet).
However I don't see the point (actually I do for FullPagePanelView, but... : ) of delegating the init work in another method while not making the initialize() really generic.
The loadtemplates should, in my opinion, be called in the OrchestraView and provided with an array of templates that would therefore become a property of the view. Therefore you no more need to manually call the initializer and loadtemplates in the children methods.

File: BackofficeBundle/Resources/public/coffee/table/TableviewView.coffee [hunk 10]
```diff
@@ -55,11 +55,21 @@ TableviewView = OrchestraView.extend(
...
       success: (response) ->
-        new FullPageFormView(viewContext.addOption(
-          html: response
-        ))
+        if panelKeys.length > 0
+          new FullPagePanelView(viewContext.addOption(
+            html: response,
+          ))
+        else
+          new FullPageFormView(viewContext.addOption(
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/table/TableviewView.coffee]: Maybe you can use there the new ViewConfigurator

File: UserAdminBundle/Controller/Admin/UserController.php [hunk 11]
```diff
@@ -66,9 +67,42 @@ public function formAction(Request $request, $userId)
...
+        ));
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $userManager = $this->get('fos_user.user_manager');
+            $userManager->updatePassword($user);
+            $this->handleForm($form, $this->get('translator')->trans('open_orchestra_user.update.success'));
+            $this->dispatchEvent(UserEvents::USER_CHANGE_PASSWORD, new UserEvent($user));
+        }
+
```
Comment (nicolasThal) [inline UserAdminBundle/Controller/Admin/UserController.php]: too many empty lines

File: UserAdminBundle/Controller/Admin/UserController.php [hunk 12]
```diff
@@ -66,9 +67,42 @@ public function formAction(Request $request, $userId)
...
+     * @param string $userId
+     *
+     * @Config\Route("/password/change/{userId}", name="open_orchestra_user_admin_user_change_password")
+     * @Config\Method({"GET", "POST"})
+     *
+     * @Config\Security("has_role('ROLE_ACCESS_USER')")
+     *
+     * @return Response
+     */
+
```
Comment (nicolasThal) [inline UserAdminBundle/Controller/Admin/UserController.php]: empty line to suppress

File: BackofficeBundle/Resources/views/BackOffice/Underscore/fullPagePanelView._tpl.twig [hunk 13]
```diff
@@ -0,0 +1,15 @@
+{% include 'OpenOrchestraBackofficeBundle:BackOffice:Underscore/Jarvis/header._tpl.twig' %}
+
+<div role="container">
+    <div id="panelView">
+        <ul class="nav nav-tabs" role="tablist" id="superboxTab">
+
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/views/BackOffice/Underscore/fullPagePanelView._tpl.twig]: These empty line is useless. Have you tried the xhtml reduced expression <ul class="..." /> ?

File: BackofficeBundle/Resources/views/BackOffice/Underscore/fullPagePanelView._tpl.twig [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+        </ul>
+        <div class="tab-content">
+
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/views/BackOffice/Underscore/fullPagePanelView._tpl.twig]: same comment here

File: UserAdminBundle/Controller/Admin/UserController.php [hunk 15]
```diff
@@ -66,9 +67,39 @@ public function formAction(Request $request, $userId)
...
+        if ($form->isValid()) {
+            $userManager = $this->get('fos_user.user_manager');
+            $userManager->updatePassword($user);
+            $this->handleForm($form, $this->get('translator')->trans('open_orchestra_user.update.success'));
+            $this->dispatchEvent(UserEvents::USER_CHANGE_PASSWORD, new UserEvent($user));
+        }
+
+        return $this->renderAdminForm($form);
+    }
+
```
Comment (nicolasThal) [inline UserAdminBundle/Controller/Admin/UserController.php]: empty line to suppress

**Comment by nicolasThal** [comment]
:+1: with the empty line suppressed

**Pull request merged**